lib/timeline/TimeStep.js

```diff
@@ -222,23 +222,7 @@ TimeStep.prototype.next = function() {
       break;
     case 'weekday':      // intentional fall through
     case 'day':          this.current.add(this.step, 'day'); break;
-    case 'week':
-      if (this.current.weekday() !== 0){ // we had a month break not correlating with a week's start before
-        this.current.weekday(0); // switch back to week cycles
-        this.current.add(this.step, 'week');
-      } else if(this.options.showMajorLabels === false) {
-        this.current.add(this.step, 'week'); // the default case
-      } else { // first day of the week
-        var nextWeek = this.current.clone();
-        nextWeek.add(1, 'week');
-        if(nextWeek.isSame(this.current, 'month')){ // is the first day of the next week in the same month?
-          this.current.add(this.step, 'week'); // the default case
-        } else { // inject a step at each first day of the month
-          this.current.add(this.step, 'week');
-          this.current.date(1);
-        }
-      }
-      break;
+    case 'week':         this.current.add(this.step, 'week'); break;
```

Comment:
I absolutely disagree with this change, which causes the first day of the month to completely disappear. In fact, I believe that showing the start of the month is mandatory, it is at least for me and so far I haven't seen anybody complaining. Week starts do not necessarily align with the start of the month. With this change, there is wrong information displayed, e.g. giving the user the impression that the month starts right with the start of the week, as is visible in your initial comment for this PR #3549, e.g. in September 2016, (ISO) week 37 started on the 12th of September. Also it is not consistent, because July looks like it was 5 weeks long. That's why I went through all the hassle with this admittedly ugly piece of code. But please leave it as it is, and maybe apply your own patch or make it configurable if you don't like it.

Reply:
I know that this changes a bit your initial implementation, but it's more generic and consistent with the rest of the code, i.e. creates a regular scale that iterates only weeks (as we are creating a week scale), without exceptions.

The rest is just a matter of label formatting, if the _majorLabel_ seems a bit misleading it can be changed for  _D MMM YYYY_ instead. But again, this can be modified using ``format: { majorLabels: {week: 'D MMM YYYY'} }``

It's not a matter of disliking the current solution, it's instead an attempt to make it more consistent with the rest of the code, usable for other cases and customizable.

Reply:
I see your point and agree that this approach is more consistent. But I'd like to suggest to at least change the major label default to D MMM YYYY if you think this change is absolutely necessary. Otherwise you get the impression of a wrong start of the month. For me, this change makes the week scale more or less useless because I really need and want to see the beginning of the month. What imho is really needed is a more flexible/independent approach to draw the major and minor labels and lines as requested in https://github.com/almende/vis/issues/3323. Please reconsider.

Reply:
I tend to agree with @marcortw . The weeks don't always align with the month so the current behavior is much preferred to me.

Reply:
Ok, I have to agreed with you guys. It's rather confusing. I'll revert this change and keep the current behavior. ;)

Can we still add the _week_ scale to the auto-scale? It could replace the 5 days scale (or keep both?) and by default its _minorLabels_ would display the "day of the month". What do you guys think?

Reply:
I'm fine with that, thanks! I initially had the weeks in the auto-scale but then left it out because I thought that people might be confused by the week number, it's probably a bit of information that is not used by many people.